File: Task/Iccanobif-primes/Python/iccanobif-primes.py
""" rosettacode.org/wiki/Iccanobif_primes """

from sympy import isprime


def iccanobifs(wanted):
    """ Print the series of iccanobif prime numbers up to wanted """
    fib, prev, prevprev, fcount = 0, 1, 0, 0
    print('First 30 Iccanobif primes:')
    while fcount < wanted:
        fib = prev + prevprev
        prevprev = prev
        prev = fib
        dig = [int(c) for c in str(fib)]
        candidate = sum(n * 10**i for i, n in enumerate(dig))
        if isprime(candidate):
            fcount += 1
            dlen = len(str(candidate))
            if dlen < 90:
                print(candidate, f"({dlen} digit{'' if dlen == 1 else 's'})")
            else:
                s = str(candidate)
                print(s[:30], "...", s[-29:], f'({dlen} digits)')


iccanobifs(30)

File: Task/Iccanobif-primes/Julia/iccanobif-primes.jl
using Primes

""" Print the series of iccanobif prime numbers up to wanted """
function iccanobifs(wanted)
    digbuf = zeros(Int, 11000)
    fib, prev, prevprev, fcount = big"0", big"1", big"0", 0
    println("First $wanted Iccanobif primes:")
    while fcount < wanted
        fib = prev + prevprev
        prevprev = prev
        prev = fib
        digits!(digbuf, fib)
        candidate = evalpoly(big"10", reverse(digbuf[begin:findlast(!iszero, digbuf)]))
        if isprime(candidate)
            fcount += 1
            dlen = ndigits(candidate)
            if dlen < 90
                println(candidate, " ($dlen digit$(dlen == 1 ? "" : "s"))")
            else
                s = string(candidate)
                println(s[1:30], " ... ", s[end-29:end], " ($dlen digits)")
            end
        end
    end
end

iccanobifs(30)
